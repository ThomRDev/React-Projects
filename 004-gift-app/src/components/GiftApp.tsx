import { useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import AddCategory from './AddCategory'
import GiftGrid from './GiftGrid'

const TitleApp = styled.h1`
    font-family: monospace;
    font-size: 4em;
`

// https://stackoverflow.com/questions/61285043/react-with-typescript-argument-of-type-never-is-not-assignable-to-parameter
const GiftApp = ({defaultValues} : { defaultValues:string[] }) => {
    const [values,setValues] = useState<string[]>([...defaultValues])
    return (
        <div >
            <TitleApp className='l-container' style={{color:"#ccc" }}>GifApp</TitleApp>
            <AddCategory setValues={setValues}  />
            <div className='l-container'>
                { values.map(value=>{
                    // return <p key={idx}>{value}</p>
                    // cuidado con el key, el orden se rendiriza feo
                    // con el key, se verficica que nos e rendice otra vez ?????
                    // o solos los nuevos valores de values se renderizan
                    return <GiftGrid key={value} value={value} /> 
                })}
            </div>
        </div>
    )
}

GiftApp.propTypes = {
    defaultValues : PropTypes.arrayOf(PropTypes.string).isRequired
}

export default GiftApp
