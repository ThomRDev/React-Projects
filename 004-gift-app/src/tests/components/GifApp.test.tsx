import { shallow } from "enzyme";
import path from "path";
import GiftApp from "../../components/GiftApp";
// https://stackoverflow.com/questions/45778192/what-is-the-difference-between-it-and-test-in-jest#:~:text=there%20is%20no%20difference.,is%20under%20the%20alias%20it%20.
describe(`Tests in ${path.basename(__filename)} file`, () => {
    test('Must render component <GifApp /> correctly', () => {
        const wrapper = shallow(<GiftApp defaultValues={ [""] } />)
        expect(wrapper).toMatchSnapshot()
    })
    test('Must render component <GifApp /> correctly with params', () => {
        const wrapper = shallow(<GiftApp defaultValues={ ["the beatles","frank sinatra"] } />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find("GiftGrid").length).toBe(2)
    })
})