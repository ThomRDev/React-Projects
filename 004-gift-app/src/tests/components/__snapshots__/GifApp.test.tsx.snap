// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifApp.test.tsx file Must render component <GifApp /> correctly 1`] = `
<div>
  <styled.h1
    className="l-container"
    style={
      Object {
        "color": "#ccc",
      }
    }
  >
    GifApp
  </styled.h1>
  <AddCategory
    setValues={[Function]}
  />
  <div
    className="l-container"
  >
    <GiftGrid
      key=""
      value=""
    />
  </div>
</div>
`;

exports[`Tests in GifApp.test.tsx file Must render component <GifApp /> correctly with params 1`] = `
<div>
  <styled.h1
    className="l-container"
    style={
      Object {
        "color": "#ccc",
      }
    }
  >
    GifApp
  </styled.h1>
  <AddCategory
    setValues={[Function]}
  />
  <div
    className="l-container"
  >
    <GiftGrid
      key="the beatles"
      value="the beatles"
    />
    <GiftGrid
      key="frank sinatra"
      value="frank sinatra"
    />
  </div>
</div>
`;
